tests/test_sami_conversion.py

```diff
@@ -7,59 +7,73 @@
     SAMPLE_SAMI, SAMPLE_SRT, SAMPLE_DFXP,
     SAMPLE_SAMI_UTF8, SAMPLE_SAMI_UNICODE, SAMPLE_DFXP_UNICODE,
     SAMPLE_SRT_UNICODE, SAMPLE_SAMI_SYNTAX_ERROR,
-    DFXP_FROM_SAMI_WITH_POSITIONING,
-    DFXP_FROM_SAMI_WITH_POSITIONING_UTF8,
-    DFXP_FROM_SAMI_WITH_POSITIONING_UNICODE,
-    SAMPLE_WEBVTT_FROM_SAMI
+    DFXP_FROM_SAMI_WITH_POSITIONING, DFXP_FROM_SAMI_WITH_POSITIONING_UTF8,
+    DFXP_FROM_SAMI_WITH_POSITIONING_UNICODE, SAMPLE_WEBVTT_FROM_SAMI,
+    SAMPLE_SAMI_PARTIAL_MARGINS, SAMPLE_SAMI_PARTIAL_MARGINS_RELATIVIZED,
+    SAMPLE_DFXP_FROM_SAMI_WITH_MARGINS, SAMPLE_SAMI_LANG_MARGIN,
+    SAMPLE_DFXP_FROM_SAMI_WITH_LANG_MARGINS
 )
 from .mixins import SRTTestingMixIn, DFXPTestingMixIn, SAMITestingMixIn, WebVTTTestingMixIn
 
+# Arbitrary values used to test relativization
+VIDEO_WIDTH = 640
+VIDEO_HEIGHT = 360
 
-class SAMIConversionTestCase(unittest.TestCase):
 
-    def setUp(self):
-        self.captions = SAMIReader().read(SAMPLE_SAMI.decode(u'utf-8'))
-        self.captions_utf8 = SAMIReader().read(SAMPLE_SAMI_UTF8.decode(u'utf-8'))
-        self.captions_unicode = SAMIReader().read(SAMPLE_SAMI_UNICODE)
-
-
-class SAMItoSAMITestCase(SAMIConversionTestCase, SAMITestingMixIn):
+class SAMItoSAMITestCase(unittest.TestCase, SAMITestingMixIn):
 
     def test_sami_to_sami_conversion(self):
-        results = SAMIWriter().write(self.captions)
+        caption_set = SAMIReader().read(SAMPLE_SAMI.decode(u'utf-8'))
```

Comment:
As discussed, at some point we'll collect all the encoding/decoding tests into their own module, because they're really their own kind of tests, and this kind of logic shouldn't really be mixed with business logic.